t/169-proxy-ssl-verify.t

```diff
@@ -0,0 +1,1179 @@
+--- grep_error_log_out
+proxy ssl verify: simple test start
+proxy ssl verify: calculated sum: 55
+proxy ssl verify: concatenated string: hello world
+proxy ssl verify: table sum: 6
+proxy ssl verify: simple test done
+
+--- no_error_log
+[error]
+[alert]
+[emerg]
+
+
+
+=== TEST 21: lua_upstream_skip_openssl_default_verify default off
+--- http_config
+    server {
+        listen unix:$TEST_NGINX_HTML_DIR/nginx.sock ssl;
+
+        ssl_certificate ../../cert/mtls_server.crt;
+        ssl_certificate_key ../../cert/mtls_server.key;
+
+        location / {
+            default_type 'text/plain';
+
+            content_by_lua_block {
+                ngx.say("lua_upstream_skip_openssl_default_verify default off")
+            }
+
+            more_clear_headers Date;
+        }
+    }
+--- config
+    location /t {
+        proxy_pass                    https://unix:$TEST_NGINX_HTML_DIR/nginx.sock;
+        proxy_ssl_verify              on;
+        proxy_ssl_name                example.com;
+        proxy_ssl_certificate         ../../cert/mtls_client.crt;
+        proxy_ssl_certificate_key     ../../cert/mtls_client.key;
+        proxy_ssl_trusted_certificate ../../cert/mtls_ca.crt;
+        proxy_ssl_session_reuse       off;
+        proxy_ssl_conf_command        VerifyMode Peer;
+
+        proxy_ssl_verify_by_lua_block {
+            ngx.log(ngx.INFO, "proxy ssl verify by lua is running!")
+        }
+    }
+--- request
+GET /t
+--- error_log
+proxy_ssl_verify_by_lua: openssl default verify
+--- no_error_log
+[error]
+[alert]
+
+
+
+=== TEST 22: lua_upstream_skip_openssl_default_verify on
+--- http_config
+    server {
+        listen unix:$TEST_NGINX_HTML_DIR/nginx.sock ssl;
+
+        ssl_certificate ../../cert/mtls_server.crt;
+        ssl_certificate_key ../../cert/mtls_server.key;
+
+        location / {
+            default_type 'text/plain';
+
+            content_by_lua_block {
+                ngx.say("lua_upstream_skip_openssl_default_verify default off")
+            }
+
+            more_clear_headers Date;
+        }
+    }
+--- config
+    location /t {
+        proxy_pass                    https://unix:$TEST_NGINX_HTML_DIR/nginx.sock;
+        proxy_ssl_verify              on;
+        proxy_ssl_name                example.com;
+        proxy_ssl_certificate         ../../cert/mtls_client.crt;
+        proxy_ssl_certificate_key     ../../cert/mtls_client.key;
+        proxy_ssl_trusted_certificate ../../cert/mtls_ca.crt;
+        proxy_ssl_session_reuse       off;
+        proxy_ssl_conf_command        VerifyMode Peer;
+
+        lua_upstream_skip_openssl_default_verify on;
+
+        proxy_ssl_verify_by_lua_block {
+            ngx.log(ngx.INFO, "proxy ssl verify by lua is running!")
+        }
+    }
+--- request
+GET /t
+--- response_body
+lua_upstream_skip_openssl_default_verify default off
```

Comment:
should be on

Reply:
The directive is `lua_upstream_skip_openssl_default_verify`, the default behavior is **NOT** to bypass openssl's default verify process, otherwise security problems maybe introduced, notice the semantic!